From 586ab5c8aa158065797e271e852e6016d4012e37 Mon Sep 17 00:00:00 2001
From: Brock Noland <brock@apache.org>
Date: Fri, 27 Feb 2015 18:36:39 +0000
Subject: [PATCH 0057/1164] CDH-25350/HIVE-9770 - Beeline ignores --showHeader
 for non-tablular output formats i.e csv,tsv,dsv
 (Naveen Gangam via Brock)

git-svn-id: https://svn.apache.org/repos/asf/hive/trunk@1662780 13f79535-47bb-0310-9956-ffa450edef68
---
 .../DeprecatedSeparatedValuesOutputFormat.java     |    5 +++++
 .../hive/beeline/SeparatedValuesOutputFormat.java  |    5 +++++
 2 files changed, 10 insertions(+)

diff --git a/beeline/src/java/org/apache/hive/beeline/DeprecatedSeparatedValuesOutputFormat.java b/beeline/src/java/org/apache/hive/beeline/DeprecatedSeparatedValuesOutputFormat.java
index f49daee..c1aea20 100644
--- a/beeline/src/java/org/apache/hive/beeline/DeprecatedSeparatedValuesOutputFormat.java
+++ b/beeline/src/java/org/apache/hive/beeline/DeprecatedSeparatedValuesOutputFormat.java
@@ -45,6 +45,11 @@ public DeprecatedSeparatedValuesOutputFormat(BeeLine beeLine, char separator) {
   public int print(Rows rows) {
     int count = 0;
     while (rows.hasNext()) {
+      if (count == 0 && !beeLine.getOpts().getShowHeader()) {
+        rows.next();
+        count++;
+        continue;
+      }
       printRow(rows, (Rows.Row) rows.next());
       count++;
     }
diff --git a/beeline/src/java/org/apache/hive/beeline/SeparatedValuesOutputFormat.java b/beeline/src/java/org/apache/hive/beeline/SeparatedValuesOutputFormat.java
index a2c18c7..75441bc 100644
--- a/beeline/src/java/org/apache/hive/beeline/SeparatedValuesOutputFormat.java
+++ b/beeline/src/java/org/apache/hive/beeline/SeparatedValuesOutputFormat.java
@@ -64,6 +64,11 @@ public int print(Rows rows) {
 
     int count = 0;
     while (rows.hasNext()) {
+      if (count == 0 && !beeLine.getOpts().getShowHeader()) {
+        rows.next();
+        count++;
+        continue;
+      }
       printRow(rows, (Rows.Row) rows.next());
       count++;
     }
-- 
1.7.9.5

